lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -214,94 +205,38 @@ def test_get_bad_query_2(self, query_get_as):
 
     def test_put_missing_json_object(self, client, server_config, pbench_token):
         """
-        Test behavior when no JSON payload is given
+        Test behavior when no dataset name is given on the URI. (NOTE that
+        Flask automatically handles this with a NOT_FOUND response.)
         """
-        response = client.put(f"{server_config.rest_uri}/datasets/metadata")
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert response.json.get("message") == "Invalid request payload"
+        response = client.put(f"{server_config.rest_uri}/datasets/metadata/")
+        assert response.status_code == HTTPStatus.NOT_FOUND
 
-    def test_put_missing_keys(self, client, server_config, pbench_token):
+    def test_put_missing_key(self, client, server_config, pbench_token):
         """
         Test behavior when JSON payload does not contain all required keys.
 
         Note that Pbench will silently ignore any additional keys that are
         specified but not required.
-
-        TODO: This is mostly copied from commons.py; ideally these would be
-        refactored into a common "superclass" analagous to ApiBase as Commons
-        is to ElasticBase.
         """
-        classobject = DatasetsMetadata(client.config, client.logger)
-
-        def missing_key_helper(keys):
-            response = client.post(
-                f"{server_config.rest_uri}/datasets/metadata",
-                json=keys,
-            )
-            assert response.status_code == HTTPStatus.BAD_REQUEST
-            missing = sorted(set(required_keys) - set(keys))
-            assert (
-                response.json.get("message")
-                == f"Missing required parameters: {','.join(missing)}"
-            )
-
-        parameter_items = classobject.schema.parameters.items()
-
-        required_keys = [
-            key for key, parameter in parameter_items if parameter.required
-        ]
-
-        all_combinations = []
-        for r in range(1, len(parameter_items) + 1):
-            for item in itertools.combinations(parameter_items, r):
-                tmp_req_keys = [key for key, parameter in item if parameter.required]
-                if tmp_req_keys != required_keys:
-                    all_combinations.append(item)
-
-        for items in all_combinations:
```

Comment:
Is this test no longer applicable or relevant?

Reply:
It was very complicated and generalized schema validation copied from `Common`; and while it made only a tiny bit of sense before when "dataset" and "metadata" were in the same space, I decided to simplify and just do them separately.

I.e., I added the `/datasets/metadata/` (and now `/datasets/metadata`) above, and this now just checks the missing `metadata` key.

Reply:
As long as we haven't lost any coverage....